### PR TITLE
Feat/periodic cleanup trash folder

### PR DESCRIPTION
**Pull request by danielchg**

### CODE REVIEW

File: cleaner/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+The script to perform the cleaning task is a Golang CLI with the below options
+
+```bash
+$ ./cleaner -h
+Usage of ./cleaner:
+  -debug
+        Set debug mode to get more detailed log of deleted files.
+  -path string
+        Specify the root path of the trash folder to be cleaned. (default "./trash")
+  -threshold int
```
Comment (guumaster) [inline cleaner/README.md:24]: Go has a native `Duration` parameters that allow to have more clear flags like: 
```
-threshold 5d  # 5 days
-threshold 3h # 3 hours
-threshold 20m  # 20 minutes
 
```

File: helm/science-toolkit/templates/common/cleaner-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+spec:
+  schedule: {{ .Values.cleaner.schedule | quote }}
+  concurrencyPolicy: Forbid
+  failedJobsHistoryLimit: 5
+  successfulJobsHistoryLimit: 2
+  startingDeadlineSeconds: 60 # 1min
+  jobTemplate:
+    spec:
+      backoffLimit: 3
+      activeDeadlineSeconds: 60 # 1min
```
Comment (guumaster) [inline helm/science-toolkit/templates/common/cleaner-cronjob.yaml:18]: Probably this process would take more than a 1 minute, is this setting ok?

File: helm/science-toolkit/values.yaml.tpl [hunk 3]
```diff
@@ -97,9 +97,15 @@ mlflow:
...
   gitea:
     enabled: false
-    schedule: "0 6 */1 * *" # every day at 6:00 AM
+    schedule: "0 1 * * 0" # every sunday at 1:00 AM
   s3:
     awsAccessKeyID: yourawskeyid
     awsSecretAccessKey: yourawssecretkey
     bucketName: yourS3BackupBucketName
 
+cleaner:
```
Comment (guumaster) [inline helm/science-toolkit/values.yaml.tpl:106]: Shouldn't the cleaner run daily and remove everything older than the threshold?

File: cleaner/main.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+var threshold int
+var trashPath string
+var debug bool
+
+func main() {
+	flag.IntVar(&threshold, "threshold", 5, "Specify the minimum age of the trash items to be removed.")
+	flag.StringVar(&trashPath, "path", "./.trash", "Specify the root path of the trash folder to be cleaned.")
+	flag.BoolVar(&debug, "debug", false, "Set debug mode to get more detailed log of deleted files.")
+	flag.Parse()
+	log.Printf("Start cleaning with the following values: \n - threshold = %v \n - trashPath = %v", threshold, trashPath)
```
Comment (guumaster) [inline cleaner/main.go:23]: You can use multiline string, also, when you know the types of the values don't use `%v` 
```
log.Printf(`Start cleaning with the following values:
  - threshold = %d
  - trashPath = %s
`, threshold, trashPath)
```

File: cleaner/main.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+	loc, _ := time.LoadLocation("UTC")
+	now := time.Now().In(loc)
+
+	// Check if trashPath exist
+	if _, err := os.Stat(trashPath); os.IsNotExist(err) {
+		log.Fatalf("The folder to clean does not exist: %v", trashPath)
```
Comment (guumaster) [inline cleaner/main.go:30]: Use `%v` for debug and types that you don't know how to print. For imple types like strings, ints and bools use the proper value `%s`, `%d`, `%t`.

File: cleaner/main.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	var itemsToRemove []string
+	if err != nil {
+		log.Fatalf("Problems listing files within trash folder: %v", err)
+	}
+	for _, trashItem := range trashItems {
+
+		fileAge := trashItem.ModTime()
+
+		if checkAgeThreshold(threshold, now, fileAge) {
+			itemsToRemove = append(itemsToRemove, fmt.Sprint(trashPath, "/", trashItem.Name()))
```
Comment (guumaster) [inline cleaner/main.go:69]: use `path.Join()` to generate valid paths for all systems

File: cleaner/main.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	if days >= threshold {
+		return true
+	}
+
+	return false
+}
+
+func listToRemove(threshold int, trashPath string, now time.Time) []string {
+	trashItems, err := ioutil.ReadDir(trashPath)
+	var itemsToRemove []string
```
Comment (guumaster) [inline cleaner/main.go:60]: change the order here, move `var` declaration one line up.

File: cleaner/main.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+		if checkAgeThreshold(threshold, now, fileAge) {
+			itemsToRemove = append(itemsToRemove, fmt.Sprint(trashPath, "/", trashItem.Name()))
+		}
+
+	}
+	return itemsToRemove
+}
+
+func removeTrashItem(itemToRemove string, wg *sync.WaitGroup) {
+
```
Comment (guumaster) [inline cleaner/main.go:77]: remove leading empty line

File: cleaner/main.go [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+			if debug {
+				log.Printf("File deleted: %v \n", info.Name())
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	os.RemoveAll(itemToRemove)
```
Comment (guumaster) [inline cleaner/main.go:94]: control error here:  `err = os.RemoveAll(itemsToRemove)`

File: cleaner/main.go [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+		if !info.IsDir() {
+			os.Remove(info.Name())
+			if debug {
+				log.Printf("File deleted: %v \n", info.Name())
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
```
Comment (guumaster) [inline cleaner/main.go:92]: DON'T PANIC :smile:

File: cleaner/main.go [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func listToRemove(threshold int, trashPath string, now time.Time) []string {
+	trashItems, err := ioutil.ReadDir(trashPath)
+	var itemsToRemove []string
+	if err != nil {
+		log.Fatalf("Problems listing files within trash folder: %v", err)
+	}
+	for _, trashItem := range trashItems {
+
```
Comment (guumaster) [inline cleaner/main.go:65]: remove leading empty line

File: cleaner/README.md (hunk 1 above)
Comment (danielchg) [reply cleaner/README.md:24]: I can't achieve use the notation 5d, only can pass 120h like time.Duration

Comment (danielchg) [reply cleaner/README.md:24]: finally we are going to use hours instead of days notation

**Pull request merged**